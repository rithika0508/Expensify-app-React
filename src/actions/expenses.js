import { v1 as uuidv1 } from 'uuid';

//add expense
export const addExpense = ({ description = '', note = '', amount = 0, createdAt = 0 } = {}) => {
    return {
        type: 'ADD-EXPENSE',
        expenses: {
            id: uuidv1(),
            description,
            note,
            amount,
            createdAt
        }
    }
}

// delete expense
export const deleteExpense = ({ id } = {}) => {
    return {
        type: 'DELETE-EXPENSE',
        id,
    };
}

// edit expense

export const editExpense = (id, updates) => {
    
    return {
        type: 'EDIT-EXPENSE',
        id,
        updates
    }
    
}