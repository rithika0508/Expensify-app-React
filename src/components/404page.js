import React from 'react';
const Notfoundpage = () => (
    <div>404 Page is not available
    </div>
);
export default Notfoundpage;