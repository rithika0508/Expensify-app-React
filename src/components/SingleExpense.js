import React from 'react';
import { NavLink } from 'react-router-dom';
export const SingleExpense = ({ id, description, note, amount, createdAt }) => {
    return (
        <div>
            <NavLink to={`/edit/${id}`}><h3>{description}</h3></NavLink>
            <p>{amount}-{createdAt}</p>
        </div>
    );
};

export default SingleExpense;