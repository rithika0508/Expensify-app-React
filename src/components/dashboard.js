import React from 'react';
import ExpenseFilters from './ExpenseFilters';
import AllExpenses from './Allexpenses';

export const ExpenseDashboardPage = (props) => {
    return (
        <div>
            <ExpenseFilters />
           <AllExpenses />
        </div>
    );
};

export default ExpenseDashboardPage;