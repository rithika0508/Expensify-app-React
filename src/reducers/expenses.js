
const expensesReducersDefaultState = []
const expensesReducers = (state = expensesReducersDefaultState, action) => {
    switch(action.type) {
        case 'ADD-EXPENSE':
            return [...state, action.expenses];
        case 'DELETE-EXPENSE':
            return state.filter((expense) => expense.id != action.id);
            case 'EDIT-EXPENSE':
                return state.map((expense) => {
                    if(expense.id == action.id) {
                        return {
                            ...expense,
                            ...action.updates,
                        }
                    } else {
                        return expense
                    }
                });
        case 'SET-EXPENSE':
            console.log(action.expenses)
            return action.expenses;
        default: 
            return state;
    }
}
export default expensesReducers;
