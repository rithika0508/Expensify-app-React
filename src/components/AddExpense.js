import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';
const AddExpense = (props) => {
    return (
        <div>
            <h2>Add Expense</h2>
            <ExpenseForm onSubmit={(expense) => {
                // props.dispatch(addExpense(expense));
                props.onSubmit(expense);
                props.history.push('/')
            }} />
        </div>
    );

};
const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (expense) => dispatch(addExpense(expense)) 
    }
}
const Connected = connect(undefined,)(AddExpense);
export default Connected;