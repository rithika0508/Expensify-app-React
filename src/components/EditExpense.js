import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense } from '../actions/expenses';
const EditExpense = (props) => {
    return (
        <div>
            This is edit expense page of id {props.match.params.id}
            <ExpenseForm expense={props.expense} onSubmit={(expense) => {
                const id=props.match.params.id;
                props.dispatch(editExpense(id, expense))
                props.history.push('/')
            }}/>
        </div>
    );
};
const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id == props.match.params.id),
        allexpenses: state.expenses
    }
}
const Connected = connect(mapStateToProps)(EditExpense)
export default Connected;