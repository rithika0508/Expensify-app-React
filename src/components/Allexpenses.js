import React from 'react';
import { connect } from 'react-redux';
import SingleExpense from './SingleExpense';
import visibleexpenses from '../selectors/expenses'


export const AllExpenses = (props) => {
    return (
        <div>
            {props.expenses.length==0 && <p>Add an expense to get started!</p>}
            {props.expenses.map((expense) => <SingleExpense key={expense.id} {...expense}/>)}
        </div>
    );
}
const mapStateToProps = (state) => {
    return {
        expenses: visibleexpenses(state.expenses, state.filter)
    }
}
const Connected = connect(mapStateToProps)(AllExpenses);
export default Connected;