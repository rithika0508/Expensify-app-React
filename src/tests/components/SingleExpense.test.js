import React from 'react';
import { SingleExpense } from '../../components/SingleExpense';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses';

test('should render single expenses', () => {
    const wrapper = shallow(<SingleExpense {...expenses[0]}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
})
