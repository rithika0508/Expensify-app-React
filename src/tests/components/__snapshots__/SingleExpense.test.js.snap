// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render single expenses 1`] = `
<div>
  <NavLink
    to="/edit/1"
  >
    <h3>
      Water Bill
    </h3>
  </NavLink>
  <p>
    500
    -
    0
  </p>
</div>
`;
